# Step 1: State the Hypotheses

Null Hypothesis (H₀): The weekly operating cost is consistent with the theoretical model (W = 1000 + 5X).

Alternative Hypothesis (H₁): The weekly operating cost is higher than the theoretical model.


In [3]:
import numpy as np
from scipy.stats import norm

# Step 2: Define the given data
sample_mean = 3050  # Sample mean weekly cost (Rs. 3,050)
sample_size = 25  # Number of restaurants in the sample
population_mean_units = 600  # Mean number of units produced (μ_X)
population_std_units = 25  # Standard deviation of units produced (σ_X)
cost_per_unit = 5  # Cost per unit (Rs. 5 per unit)
fixed_cost = 1000  # Fixed cost (Rs. 1,000)

# Step 3: Calculate the theoretical mean weekly cost (μ) using the cost model
# Theoretical cost model: W = 1000 + 5X
theoretical_mean_cost = fixed_cost + cost_per_unit * population_mean_units
print(f"Theoretical Mean Weekly Cost (μ): Rs. {theoretical_mean_cost}")

# Step 4: Calculate the standard deviation of the weekly cost (σ_W)
# Since W = 1000 + 5X, the standard deviation of W is 5 * σ_X
std_weekly_cost = cost_per_unit * population_std_units
print(f"Standard Deviation of Weekly Cost (σ_W): Rs. {std_weekly_cost}")

# Step 5: Calculate the standard error of the mean (SEM)
# SEM = σ_W / sqrt(n)
sem = std_weekly_cost / np.sqrt(sample_size)
print(f"Standard Error of the Mean (SEM): Rs. {sem}")

# Step 6: Calculate the test statistic (Z-score)
# Z = (sample_mean - theoretical_mean_cost) / SEM
z_score = (sample_mean - theoretical_mean_cost) / sem
print(f"Test Statistic (Z-score): {z_score}")

# Step 7: Determine the critical value for alpha = 0.05 (one-tailed test)
alpha = 0.05  # Significance level
critical_value = norm.ppf(1 - alpha)  # Critical value from the standard normal distribution
print(f"Critical Value (Z-critical): {critical_value}")

# Step 8: Make a decision
if z_score > critical_value:
    decision = "Reject the null hypothesis: There is strong evidence that weekly operating costs are higher."
else:
    decision = "Fail to reject the null hypothesis: There is no strong evidence that weekly operating costs are higher."

# Step 9: Print the conclusion
print("\nConclusion:")
print(decision)

Theoretical Mean Weekly Cost (μ): Rs. 4000
Standard Deviation of Weekly Cost (σ_W): Rs. 125
Standard Error of the Mean (SEM): Rs. 25.0
Test Statistic (Z-score): -38.0
Critical Value (Z-critical): 1.6448536269514722

Conclusion:
Fail to reject the null hypothesis: There is no strong evidence that weekly operating costs are higher.


# Conclusion:

The Z-score (-38.0) is much less than the critical value (1.6449), so we fail to reject the null hypothesis. There is no strong evidence to support the restaurant owners' claim that the weekly operating costs are higher than the theoretical model suggests.